reverseforwardrecord/get_dns_record.py

```diff
@@ -0,0 +1,255 @@
+import sys
+from typing import List, Dict, Tuple
+from functools import reduce
+import socket
+import pandas as pd
+import argparse
+import subprocess
+
+"""
+The IP Resolver script is a utility designed to validate and resolve IP addresses and DNS records
+from provided data files. The script supports input files in both XLSX and CSV formats,
+containing specific columns for Fully Qualified Domain Names (FQDN) and IDRAC IP addresses.
+Using these inputs, the script ensures that the IP addresses and DNS records are correctly
+formatted and reachable.
+"""
+
+
+def create_file(
+        out_filepath: str = "output.txt"
+):
+    """
+    makes a txt file for the results to be returned into
+    :return: output.txt
+    """
+    with open(out_filepath, "w"):
+        pass
+
+
+def read_from_netbox(
+        netbox_filepath: str, fqdn_column: str = "FQDN", idrac_ip_column: str = "IDRAC IP"
+) -> List[Dict]:
+    """
+    reads from netbox file, in a xlsx format
+    :param netbox_filepath: path for the netbox info
+    :param fqdn_column: full qualified domain name column in netbox file
+    :param idrac_ip_column: idrac ip column name in netbox file
+    :return: a list of dictionaries containing fqdn and idrac_ip pairs
+    """
+
+    sheetnames = pd.ExcelFile(netbox_filepath).sheet_names
+
+    # get all sheets on the XLSX file.
+    workbook_info = []
+
+    for i, sheet in enumerate(sheetnames, 1):
+        workbook = pd.read_excel(netbox_filepath, sheet_name=sheet, engine='openpyxl').to_dict()
+
+        if fqdn_column not in workbook or idrac_ip_column not in workbook:
+            raise RuntimeError(
+                f"Error: you've provided an excel sheet: (no. {i}) which does not contain one or more of these "
+                f"following column in workbook: {fqdn_column}, {idrac_ip_column}"
+            )
+        for key, fqdn_val in workbook[fqdn_column].items():
+            workbook_info.extend([
+                {
+                    "fqdn": fqdn_val,
+                    "idrac_ip": workbook[idrac_ip_column][key]
+                }
+            ])
+    return workbook_info
+
+
+def parse_netbox_info(
+        netbox_info: List[Dict], reverse_order: bool = False
+) -> List[Dict]:
+    """
+    Parse info from netbox and create corresponding hypervisor fqdn and ip value
+    :param netbox_info: list of dictionaries holding fqdn and idrac_ip
+    :param reverse_order: use reverse or forward order / if true use forward if  false use reverse
+    :return: list of dictionaries which will contain hypervisor fqdn and ip keys-value pairs
+    """
+    parsed_info = []
+    for item in netbox_info:
+        parsed_item = {}
+        if reverse_order:
+            idrac = reduce(
+                lambda x, y: f"{x}.{y}", item["idrac_ip"].split(".")[::-1]
+            )  #
+            idrac = idrac.rsplit(".", 1)[0]
```

Comment:
here's a much cleaner way to do the same thing
```suggestion
            ip_components = item["idrac_ip"].split(".")
            ip_components = ip_components[:1][::-1] # remove first octet and reverse ip
            idrac = '.'.join(ip_components) 
```

Reply:
doesn't work replaces all the ips with the number 10